### DISCUSSION TITLE
Zero crossing ?

### DISCUSSION THREAD

**Discussion by Inssomniak** [pschatzmann/arduino-audio-tools#1007] [Q&A] [answered]
Hello

Is there a facility in any of the audio tools that one can use analog audio ADC or an i2s stream to determine zero crossing of the audio ? I’m looking for a way to run_some_code when the waveform is at zero/silent. 

Thanks!

**Comment by pschatzmann** [comment, marked as answer]
I am not really sure what your purpose is. Though I am not officially supporting this I suggest that you have a look at the FrequncyZeroCrossingStream class. You can register a callback to be informed for each upward 0 crossing.

**Reply by Inssomniak** [reply]
Thanks that is exactly what I’m looking for. The purpose is to send commands to digital potentiometers when the audio output is at or near 0 to eliminate audio glitches.
